### ISSUE TITLE
Define macro directive

### ISSUE THREAD

**Issue by davschneller** [TUM-LRR/era-gp-sim#78]

**Comment by kreudom** [comment]
Hi,
the changes look good, tests are running fine as well. Implementation should be easy with this, thanks for your good work!

**Comment by andizimmerer** [comment]
As always there are no big issues :)
But I'd like to mention some points:
- `determineMemoryPosition()`: Will it stay empty and why exactly is this dummy method there?
- Error messages regarding macros (intermediate-representator.hpp:62/71): I would rephrase the messages because they were not clear to me on the first glance. You could simply add "... nested macros are not supported". This makes it easier to understand. 
  But anyway: well done!

**Comment by davschneller** [comment]
Thanks alot you two! I have changed the thing with the compile error messages.
And yes, `determineMemoryPosition` stays empty for this branch. There will be probably another issue where this might be fixed.
I'm gonna close it for now, if you are not satisfied with the error messages yet or have some other complaints, feel free to reopen the issue.
